src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.cwrdapi.servicebus;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jms.IllegalStateException;
+import org.springframework.jms.connection.CachingConnectionFactory;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import javax.jms.ConnectionFactory;
+import javax.jms.Session;
+
+@Slf4j
+@Service
+public class TopicPublisher {
+
+    private final JmsTemplate jmsTemplate;
+    private final String destination;
+    private final ConnectionFactory connectionFactory;
+
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
+    @Autowired
+    public TopicPublisher(JmsTemplate jmsTemplate,
+                          @Value("${amqp.topic}") final String destination,
+                          ConnectionFactory connectionFactory) {
+        this.jmsTemplate = jmsTemplate;
+        this.destination = destination;
+        this.connectionFactory = connectionFactory;
+    }
+
+    @Retryable(maxAttempts = 5, backoff = @Backoff(delay = 2000, multiplier = 3))
+    public void sendMessage(final String message) {
+        log.info("{}:: Sending message.", loggingComponentName);
+        try {
+            jmsTemplate.send(destination, (Session session) -> session.createTextMessage(message));
+            log.info("{}:: Message sent.", loggingComponentName);
+        } catch (IllegalStateException e) {
+            if (connectionFactory instanceof CachingConnectionFactory) {
+                log.info("{}:: Send failed, attempting to reset connection...", loggingComponentName);
+                ((CachingConnectionFactory) connectionFactory).resetConnection();
```

Comment:
ideally catch in exception rather than looping if ..else. And throw custom runtime exception. Create you own and throw that with message from coming exception.

Reply:
Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/59/commits/850ac5c71559686b1bc928ea25a40d7202fd9b53